tests/test_eligibility.py

```diff
@@ -357,19 +353,21 @@ def _do_test(country, council_tax_band, epc_rating):
     page = form.submit().follow()
 
     data = interface.api.session.get_answer(session_id, page_name="address")
-    assert data["building_name_or_number"] == "10"
-    assert data["postcode"] == "SW1A 2AA"
+    # assert data["building_name_or_number"] == "10"
+    # assert data["postcode"] == "SW1A 2AA"
```

Comment:
Be careful not to commit commented out code! I'm assuming that these assertions need to be reinstated.

Reply:
I've removed these assertions as they no longer appear in the data associated with the answer. I'm not sure that's actually best practice. In `views.py`, in the handle_post method for the `AddressView`, the `building_name_or_number` and `postcode` that the user has provided are used to make a call to the EPC API; however, these two variables are overwritten by the response from the API:
```
        building_name_or_number = data["building_name_or_number"]
        postcode = data["postcode"]
        country = interface.api.session.get_answer(session_id, "country")["country"]
        try:
            if country == "Scotland":
                return redirect("frontdoor:page", session_id=session_id, page_name="address-select")
            else:
                address_and_rrn_details = interface.api.epc.get_address_and_epc_rrn(building_name_or_number, postcode)
                interface.api.session.save_answer(
                    session_id, page_name, {"address_and_rrn_details": address_and_rrn_details}
                )
```

Would it be better to keep all the data in there as a record of what the user has actually inputted on this page?

Reply:
Aha, thanks for explaining!

Looking into the tests further, the other relevant point is that you are adding back the other assertions (as discussed below):
```
    assert data["rrn"] == "1111-1111-1111-1111-1111"
    assert data["address"] == "10, DOWNING STREET, LONDON, CITY OF WESTMINSTER, SW1A 2AA"
```

Presumably the `address`, those would only be correctly populated if both the `building_name_or_number` and the `postcode` were correct. So we are testing the handling of those inputs anyway.

That being the case, I think removing these assertions is the correct thing to do, as you suggest. We don't need to keep the data around for any purpose.